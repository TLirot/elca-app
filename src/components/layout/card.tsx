import React, { FC, ReactElement, ReactNode, useMemo } from "react";

export interface CardProps {
    children: ReactNode,
    title: ReactNode,
}

export const Card: FC<CardProps> = (props): ReactElement => {
    const { children, title } = props;

    const cardTitle = useMemo(() => {
        return title ? (
            <div className={"card-title"}>
                {title}
            </div>
        ) : null
    }, [ title ])

    return (
        <div className={"card"}>
            <div className={"card-body"}>
                {cardTitle}
                {children}
            </div>
        </div>
    )
}
