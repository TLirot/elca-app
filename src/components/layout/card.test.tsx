import { Card, CardProps } from "./card";
import { shallow, ShallowWrapper } from "enzyme";

const defaultProps: CardProps = {
    children: (<div>children</div>),
    title: "my title",
}
const initShallowComponent = (customProps: Partial<CardProps> = {}): ShallowWrapper => {

    const { children, ...restProps } = {
        ...defaultProps,
        ...customProps,
    };

    return shallow(<Card {...restProps}> {children} </Card>);
}

describe("Card =>", () => {
    beforeEach(() => {
        jest.restoreAllMocks();
    });

    describe("render =>", () => {
        it("Should render the title and the children.", () => {
            expect(initShallowComponent()).toMatchSnapshot();
        });

        it("Should render only the children.", () => {
            expect(initShallowComponent({ title: undefined })).toMatchSnapshot();
        });
    });
});
