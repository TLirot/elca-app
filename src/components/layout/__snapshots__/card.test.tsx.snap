// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card => render => Should render only the children. 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
     
    <div>
      children
    </div>
     
  </div>
</div>
`;

exports[`Card => render => Should render the title and the children. 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title"
    >
      my title
    </div>
     
    <div>
      children
    </div>
     
  </div>
</div>
`;
